public class ArrayIndexOutOfBoundsException2 {
    //@ requires size >= 0;
    //@ ensures \result == 1 ==> (\exists int i; 0 <= i && i < 4; i == size);
    //@ ensures \result == 0 ==> (\forall int i; 0 <= i && i < 4; i != size);
    public static int func(int size) {
        if (size < 0) return -1;
        try {
            int[] a = new int[4];
            int i = a[size];
        } catch (ArrayIndexOutOfBoundsException exc) {
            return 0;
        }
        return 1;
    }
}

